File: Models/Authentication/Password/ResetPassword.cs
namespace aliment_backend.Models.Authentication.Password
{
    /// <summary>
    /// Représente les informations nécessaires pour réinitialiser le mot de passe d'un utilisateur.
    /// </summary>
    public class ResetPassword
    {
        /// <summary>
        /// Obtient ou définit le nouveau mot de passe pour l'utilisateur.
        /// </summary>
        public CreatePassword? Password { get; set; }

        /// <summary>
        /// Obtient ou définit l'adresse e-mail de l'utilisateur dont le mot de passe doit être réinitialisé.
        /// </summary>
        public string? Email { get; set; }

        /// <summary>
        /// Obtient ou définit le jeton de réinitialisation du mot de passe.
        /// </summary>
        public string? Token { get; set; }
    }
}

File: src/app/features/auth/_models/resetPassword.ts
import { CreatePassword } from "./createPassword";

/**
 * L'interface `ResetPassword` est utilisée pour représenter les informations nécessaires à la réinitialisation d'un mot de passe.
 * Elle inclut les données du nouveau mot de passe (CreatePassword), l'adresse e-mail associée au compte, et le jeton de réinitialisation.
 * Elle est généralement utilisée dans le cadre d'une fonctionnalité de réinitialisation de mot de passe.
 */
export interface ResetPassword {
  /**
   * Nouveau mot de passe à définir lors de la réinitialisation.
   * @type {CreatePassword}
   */
  password: CreatePassword;

  /**
   * Adresse e-mail associée au compte pour la réinitialisation du mot de passe.
   * @type {string}
   */
  email: string;

  /**
   * Jeton de réinitialisation du mot de passe.
   * @type {string}
   */
  token: string;
}
